lib/App/MintTag.pm

```diff
@@ -493,6 +495,42 @@ sub _maybe_delete_source_branches ($self, $step) {
   }
 }
 
+sub _maybe_cleanup_tags ($self, $step) {
+  return unless ($step->push_tag_to && $step->cleanup_tag_days);
+
+  my $remote = $step->push_tag_to;
+  my $prefix = $step->tag_prefix;
+
+  try {
+    my $tag_spec = "refs/tags/${prefix}*";
+    my $tag_list = run_git('for-each-ref', "--format=%(refname:short)\t%(creatordate:unix)", $tag_spec);
+
+    my @tags = map {; my ($tag, $ts) = split /\t/ ; { ts => $ts, 'tag' => $tag}  }
+    split /\r?\n/, $tag_list;
+
+    my $time = time;
+
+    my $max_ts = $time - ($step->cleanup_tag_days * 86400);
+
+    for my $chunk (@tags) {
```

Comment:
This loop will go much faster if you do this instead:
```perl
my @can_remove = map {; $_->{tag} } grep {; $_->{ts} < $max_ts } @chunks;
# logging goes here
git_run('push', $remote->name, map {; ":refs/tags/$_" } @can_remove);
```

That wild do one network round trip instead of N.

In theory, you could end up with more than the maximum number of arguments, but I would probably not worry about it.  If you want to worry about it, you'd say:

```perl
my @can_remove = (...);
while (my @batch = splice @can_remove, 0, 100) {
  git_run(..., map {...} @batch);
}
```

or something, so it's not absurd to do if you'd be happier.